src/plugins/intel_gpu/src/graph/reshape.cpp

```diff
@@ -115,7 +115,11 @@ layout reshape_inst::calc_output_layout(reshape_node const& node, kernel_impl_pa
     if (desc->output_shape.count() == 0) {
         if (desc->output_partial_shape.size() != 0) {
             format out_fmt = format::adjust_to_rank(input_layout.format, desc->output_partial_shape.rank().get_length());
-            return layout{desc->output_partial_shape, input_layout.data_type, out_fmt};
+            auto use_new_shape_infer = node.get_program().is_new_shape_infer();
+            auto output_shape = desc->output_partial_shape;
+            if (!use_new_shape_infer && output_shape.size() < 4)
```

Comment:
Why do we need to exclude dynamic shapes? Is there any specific reason for that?

Reply:
Because dynamic shape uses partial_shape while legacy static shape uses tensor (4d aligned dims). We don't need to align partial shape.